### DISCUSSION TITLE
tsx: command not found (MacOS)

### DISCUSSION THREAD

**Discussion by zookil123** [mayooear/ai-pdf-chatbot-langchain#211] [Q&A]
Received after ingest command, `npm run ingest`:

> gpt4-langchain-pdf-chatbot@0.1.0 ingest
> tsx -r dotenv/config scripts/ingest-data.ts
sh: tsx: command not found

FIX: 
ensure that you have ts-node and typescript installed as dev dependencies in your project:

`npm install --save-dev ts-node typescript`

**Comment by Aincvy** [comment]
~~In my case, install this package: https://www.npmjs.com/package/tsx~~

Update to lastest node, npm and yarn, then yarn install, then the question is gone.

**Comment by YIN-Renlong** [comment]
Thanks for above answers, however, for those who are stuck executing the command `npm install --save-dev ts-node typescript` for a long time may try this method:

`npm install --save-dev ts-node typescript --registry https://registry.yarnpkg.com`
